### PR TITLE
Add base spider for doem.org.br and spider for Acajutiba/BA

### LINKED ISSUE #300: Add base spider for Diário Oficial Eletrônico dos Municípios
Different cities are under http://doem.org.br/ domain. We could benefit from it having a base spider for this source.

Examples (only from Bahia):

* Acajutiba BA
* Alagoinhas BA
* Cachoeira BA
* Campo Formoso BA
* Catu BA
* Ipiaú BA
* Itapetinga BA
* Juazeiro BA
* Mucuri BA
* Porto Seguro BA
* Ribeira do Pombal BA
* Santo Amaro BA
* Senhor do Bonfim BA
* São Gonçalo dos Campos BA
* Tucano BA

### PR DESCRIPTION
**Pull request by rodbv**
Closes #300 

This is a new PR due to the several changes done on the project structure, check PR #330 for reference.

### CODE REVIEW

File: data_collection/gazette/spiders/base.py [hunk 1]
```diff
@@ -188,3 +188,85 @@ def get_gazette(self, document):
...
+            edition_number = self.get_edition_number(gazette_box)
+            yield Gazette(
+                date=date,
+                file_urls=[file_url],
+                edition_number=edition_number,
+                power="executive_legislative",
+            )
+
+    def get_url(self, page=None):
+        state, city = self.name.split("_")
```
Comment (giuliocc) [inline data_collection/gazette/spiders/base.py:233]: Just a comment about cities with multiword names, they don't use the same pattern we use here. All words are joined together without `"_"`. Consider if you want to change it now or change it later.

One thing to think about: I personally don't like to rely on the spider name to make anything in the crawl. This is mainly by one bad experience after changing naming patterns for the project and having some spiders break down unnecessarily. For me, making a minor "code repetition" is valid. What do you think?

Comment (rodbv) [reply data_collection/gazette/spiders/base.py:233]: Excelent points. Explicit is better than implicit :) 
I'll make the city name as expected to build the URL an attribute of the spider instance, to be provided by the person building the specific child-spider.

**Comment by rodbv** [comment]
From the 200 or so cities in  DOEM, I've checked  which ones are actually using the platform currently, and for those I've created 40 spiders that will use the base spider from this PR: https://github.com/rodbv/doem_spider_generator/tree/main/2020_11_05_1604613250

File: data_collection/gazette/spiders/base.py [hunk 2]
```diff
@@ -188,3 +188,84 @@ def get_gazette(self, document):
...
         )
+
+
+class DoemGazetteSpider(BaseGazetteSpider):
+    """
+    Base spider for all cities listed on https://doem.org.br
+    """
+
+    allowed_domains = ["doem.org.br"]
+    start_date = date(2009, 1, 1)
```
Comment (jvanz) [inline data_collection/gazette/spiders/base.py:199]: Is this valid for all cities?

File: data_collection/gazette/spiders/base.py [hunk 3]
```diff
@@ -188,3 +188,84 @@ def get_gazette(self, document):
...
+            scrapy.Request(self.get_url(page), callback=self.parse)
+            for page in range(1, 1 + self.get_last_page(response))
+        ]
+
+    def parse(self, response):
+        """
+        Parse each page from the results page and yield the gazette issues available.
+        """
+        GAZETTE_CONTAINER_CSS_QUERY = "div.box-diario"
+        gazette_boxes = response.css(GAZETTE_CONTAINER_CSS_QUERY)
```
Comment (jvanz) [inline data_collection/gazette/spiders/base.py:219]: Nice! I really like meaningful names for selectors. My human brain appreciates

**Review by jvanz — APPROVED** [review]
Small and succinct functions. I love it. Thanks!

File: data_collection/gazette/spiders/base.py (hunk 2 above)
Comment (rodbv) [reply data_collection/gazette/spiders/base.py:199]: Yes, as far as I could see they're all after 2013.

**Review by giuliocc — APPROVED** [review]

**Comment by giuliocc** [comment]
Great work, @rodbv ! Paving the way for 40 more cities ;)

**Pull request merged**